draft-ietf-quic-transport.md

```diff
@@ -4584,13 +4584,14 @@ stateless_reset_token (0x0002):
   reset ({{stateless-reset}}) for the connection ID negotiated during the
   handshake.
 
-max_packet_size (0x0003):
+max_udp_size (0x0003):
 
-: The maximum packet size parameter is an integer value that limits the size of
-  packets that the endpoint is willing to receive.  This indicates that packets
-  larger than this limit will be dropped.  The default for this parameter is the
-  maximum permitted UDP payload of 65527.  Values below 1200 are invalid.  This
-  limit only applies to protected packets ({{packet-protected}}).
+: The maximum udp size parameter is an integer value that limits the size of
+  UDP payloads that the endpoint is willing to receive.  This indicates that
+  packets larger than this limit are likely not to be processed by the receiver.
```

Comment:
I would change this sentence to something like `This indicates that UDP packets whose payload is larger than this limit are likely not to be processed by the receiver.`

Reply:
agreed.

Reply:
I think we need to talk about both UDP payload size and QUIC packet size, because they are different things.

I would assume that we'd all agree that an endpoint is allowed to process UDP datagrams that are larger than the advertised maximum. The endpoint cannot consider that as a protocol violation, because UDP datagrams are not authenticated.

Contrary to that, certain QUIC packets are "protected," and an endpoint _can_ report a protocol violation if it receives a protected QUIC packet that can be decrypted, but is larger than the advertised maximum. If such check needs to be done (or should not be done) is under debate in #3447.

The new text says that "packets larger than this limit are likely not to be processed."

I'd interpret this as "a receiver MAY process packets larger than this limit", and therefore would consider that this PR at the moment is trying to also close #3447, by implying that "a receiver MUST NOT close the connection when it receives a QUIC packet that is greater than the advertised limit."

Reply:
Are we really concerned about nodes performing this check? My understanding is that we added this transport parameter in the first place to allow folks to read UDP into a fixed-size stack buffer, and this allows an endpoint to communicate the size of that buffer to its peer. If I'm able to decrypt this packet it means I'm able to read and process it, so I shouldn't be sending a lower transport parameter. I think the sentence I propose above makes it clear that this is advisory and should not be enforced.

Reply:
@DavidSchinazi I totally agree with your rationale. However, #3447 discusses about enforcing the maximum at the QUIC packet level.

Anyways, regarding enforcement, I think what we need to build consensus, before discussing how it should be baked into the specification. And that might be out-of-scope for a PR that aims to be editorial. Based on that view, I think it is fine to just talk about UDP payload size here, and that having only the text you proposed is fine.

Reply:
I've applied @DavidSchinazi's original rewording suggestion but am leaving this conversation open as the subject has drifted off.